### PR TITLE
[core] Introduce View Support to HiveCatalog

### PR DESCRIPTION
**Pull request by JingsongLi**
### Purpose

This PR aims to support View in Catalog, our metastore behind us, such as Hive metastore, supports views, but our catalog has not done this conversion well. This PR aims to support the definition of views and bridge with HiveCatalog.

#### Definition of view

```
/** Interface for view definition. */
public interface View {

    /** A name to identify this view. */
    String name();

    /** Full name (including database) to identify this view. */
    String fullName();

    /** Returns the row type of this view. */
    RowType rowType();

    /** Returns the view representation. */
    String query();

    /** Optional comment of this view. */
    Optional<String> comment();

    /** Options of this view. */
    Map<String, String> options();

    /** Copy this view with adding dynamic options. */
    View copy(Map<String, String> dynamicOptions);
}
```

#### Catalog interfaces

```
/**
     * Return a {@link View} identified by the given {@link Identifier}.
     *
     * @param identifier Path of the view
     * @return The requested view
     * @throws ViewNotExistException if the target does not exist
     */
    default View getView(Identifier identifier) throws ViewNotExistException {
        throw new ViewNotExistException(identifier);
    }

    /**
     * Drop a view.
     *
     * @param identifier Path of the view to be dropped
     * @param ignoreIfNotExists Flag to specify behavior when the view does not exist: if set to
     *     false, throw an exception, if set to true, do nothing.
     * @throws ViewNotExistException if the view does not exist
     */
    default void dropView(Identifier identifier, boolean ignoreIfNotExists)
            throws ViewNotExistException {
        throw new UnsupportedOperationException();
    }

    /**
     * Create a new view.
     *
     * @param identifier path of the view to be created
     * @param view the view definition
     * @param ignoreIfExists flag to specify behavior when a view already exists at the given path:
     *     if set to false, it throws a ViewAlreadyExistException, if set to true, do nothing.
     * @throws ViewAlreadyExistException if view already exists and ignoreIfExists is false
     * @throws DatabaseNotExistException if the database in identifier doesn't exist
     */
    default void createView(Identifier identifier, View view, boolean ignoreIfExists)
            throws ViewAlreadyExistException, DatabaseNotExistException {
        throw new UnsupportedOperationException();
    }

    /**
     * Get names of all views under this database. An empty list is returned if none exists.
     *
     * @return a list of the names of all views in this database
     * @throws DatabaseNotExistException if the database does not exist
     */
    default List<String> listViews(String databaseName) throws DatabaseNotExistException {
        throw new UnsupportedOperationException();
    }
```

Implemented these methods in `HiveCatalog`.

#### Flink Integration

Integrate to Flink Catalog.

### CODE REVIEW

**Comment by adrian-wang** [comment]
maybe also support `listViews` and `alterView`?

File: paimon-core/src/main/java/org/apache/paimon/view/ViewImpl.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    @Override
+    public View copy(Map<String, String> dynamicOptions) {
+        Map<String, String> newOptions = new HashMap<>(options);
+        newOptions.putAll(dynamicOptions);
+        return new ViewImpl(identifier, rowType, query, comment, newOptions);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
```
Comment (wwj6591812) [inline paimon-core/src/main/java/org/apache/paimon/view/ViewImpl.java:92]: NeedBraces: 'if' construct must use '{}'s.

File: paimon-core/src/main/java/org/apache/paimon/view/ViewImpl.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        if (o == null || getClass() != o.getClass()) return false;
```
Comment (wwj6591812) [inline paimon-core/src/main/java/org/apache/paimon/view/ViewImpl.java:93]: NeedBraces: 'if' construct must use '{}'s.

**Comment by JingsongLi** [comment]
> maybe also support `listViews` and `alterView`?

Alter view looks invalid to me, it just modify properties, I cannot see the case.

**Comment by LinMingQiang** [comment]
Is there any problem with this PR? I am looking forward to this feature.

**Comment by adrian-wang** [comment]
LGTM

**Comment by JingsongLi** [comment]
> Is there any problem with this PR? I am looking forward to this feature.

Let's merge this.

**Pull request merged**